cmdstanpy/utils/stancsv.py

```diff
@@ -129,245 +134,384 @@ def parse_hmc_adaptation_lines(
             break
         elif b"diag_e" in line:
             diag_e_metric = True
-    if step_size is None:
-        raise ValueError("Unable to parse adapated step size")
     if matrix_lines:
-        mass_matrix = csv_bytes_list_to_numpy(
-            matrix_lines, includes_header=False
-        )
+        mass_matrix = csv_bytes_list_to_numpy(matrix_lines)
         if diag_e_metric and mass_matrix.shape[0] == 1:
             mass_matrix = mass_matrix[0]
     return step_size, mass_matrix
 
 
+def extract_key_val_pairs(
+    comment_lines: List[bytes], remove_default_text: bool = True
+) -> Iterator[Tuple[str, str]]:
+    """Yields cleaned key = val pairs from stan csv comments.
+    Removes '(Default)' text from values if remove_default_text is True."""
+    cleaned_lines = (
+        line.decode().lstrip("# ").strip() for line in comment_lines
+    )
+    for line in cleaned_lines:
+        split_on_eq = line.split(" = ")
+        # Only want lines with key = value
+        if len(split_on_eq) != 2:
+            continue
+
+        key, val = split_on_eq
+        if remove_default_text:
+            val = val.replace("(Default)", "").strip()
+        yield key, val
+
+
+def parse_config(
+    comment_lines: List[bytes],
+) -> Dict[str, Union[str, int, float]]:
+    """Extracts the key=value config settings from Stan CSV comment
+    lines and returns a dictionary."""
+    out: Dict[str, Union[str, int, float]] = {}
+    for key, val in extract_key_val_pairs(comment_lines):
+        if key == 'file':
+            if not val.endswith('csv'):
+                out['data_file'] = val
+        else:
+            if val == 'true':
+                out[key] = 1
+            elif val == 'false':
+                out[key] = 0
+            else:
+                for cast in (int, float):
+                    try:
+                        out[key] = cast(val)
+                        break
+                    except ValueError:
+                        pass
+                else:
+                    out[key] = val
```

Comment:
Could this just be `for cast in (bool, int, float)` instead of special casing `true` and `false` like this?

Reply:
Oh, probably not because they're not capitalized?